src/array/ops/utf8.rs

```diff
@@ -0,0 +1,106 @@
+use crate::array::BaseArray;
+use crate::datatypes::{BooleanArray, Utf8Array};
+use arrow2::array as arrow2_array;
+
+use crate::error::{DaftError, DaftResult};
+
+impl Utf8Array {
+    pub fn endswith(&self, pattern: &Utf8Array) -> DaftResult<BooleanArray> {
+        let data_arrow = self.downcast();
+        let pattern_arrow = pattern.downcast();
+        let result_arrow = match (self.len(), pattern.len()) {
+            // Broadcast pattern case:
+            (data_len, 1) => match pattern_arrow.validity() {
+                Some(validity) if !validity.get_bit(0) => Ok(arrow2_array::BooleanArray::new_null(
```

Comment:
if you are okay returning the DataArray directly from here: you can use `full_null`: https://github.com/Eventual-Inc/Daft/blob/62ac0c16e0294a709dc36d1044cab83354cb31a9/src/array/ops/full.rs#L15